app/controllers/projects_controller.rb

```diff
@@ -27,16 +27,24 @@ def update
     end
   end
 
-  def create
-    @project = current_user.projects.new(project_params)
-    project_name = @project.project_name
-
+  # returns true if ok, false if not
+  def valid_project_name?(project_name)
     if project_name.nil? or project_name.empty?
       flash[:message] = "Invalid project name"
-      redirect_to new_project_path and return
+      return false
     elsif !Project.where(
         project_name: project_name, user_id: @project.user_id).blank?
       flash[:message] = "Project name already exists"
+      return false
+    end
+    true
```

Comment:
nit but why not 'return true' since you went with it in lines 34/38

Reply:
RubyMine gave me an "unnecessary return statement" warning, so I removed it. I think it's probably a Ruby style convention that if it's the last evaluated line then we should remove the "return" keyword. Whereas the previous lines wouldn't be the last evaluated lines without the return keyword (and would go to the `return true`). So the inconsistency might be a Ruby recommended style lol.